### ISSUE TITLE
Add mime type inference when uploaded images are missing mime type

### ISSUE THREAD

**Issue by ajbozarth** [IBM/MAX-Image-Caption-Generator#13]
#9 added error checking for file type, but did so using mime type. This means that uploads with missing mime type (mime type = `None`) will fail even if the image is valid. This was discovered and addressed by the web app in https://github.com/IBM/MAX-Image-Caption-Generator-Web-App/pull/39

As per discussion in #9 we should look into adding the mime type fix in the model as well. This can either be done using the same method as the web app or by using the more robust python magic library.

**Comment by gdequeiroz** [comment]
@ajbozarth should this be closed via #9?

**Comment by ajbozarth** [comment]
No this is a follow up to #9 but as referenced there we can close this as "will not fix" if we would rather not spend the dev time. This is a minor edge case that can be worked around client side when encountered (like we did in the web app).

**Comment by xuhdev** [comment]
Yes, I agree. A properly written web application should send the correct mimetype instead of compromising the stability of this model.
